yoyodyne/util.py

```diff
@@ -24,7 +24,39 @@ def __call__(
         getattr(namespace, self.dest).add(values)
 
 
-# Tensor manipulation
+def log_info(msg: str) -> None:
+    """Logs msg to sys.stderr.
+
+    We can additionally consider logging to a file, or getting a handle to the
+    PL logger.
+
+    Args:
+        msg (str): the message to log.
+    """
+    print(msg, file=sys.stderr)
+
+
+def log_arguments(args: argparse.Namespace) -> None:
+    """Logs non-null arguments via log_info.
+
+    Args:
+        args (argparse.Namespace).
+    """
+    log_info("Arguments:")
+    for arg, val in vars(args).items():
+        if val is None:
+            continue
+        log_info(f"\t{arg}: {val!r}")
+
+
+def mkpath(path: str) -> None:
+    """Creates subdirectories for a path if they do not already exist.
+
+    Args:
+        path (str).
+    """
+    dirname = os.path.dirname(os.path.abspath(path))
```

Comment:
Is this how we created paths for artifacts previously? If not, is there a reason to prefer abs over rel paths?

Reply:
No, I am being more perspicuous than before.

`abspath` is the standard way to get Python to correctly expand stuff like `..` in an OS-specific way; `relpath` doesn't do any such thing.